### ISSUE TITLE
"Cannot read property 'extend' of undefined" with Next.js

### ISSUE THREAD

**Issue by gaku-sei** [grommet/grommet#3497]
When used with Next.js 9.1.0, the `FormField` component raises a `Cannot read property 'extend' of undefined` error on the server side.

### Actual Behavior

When using Grommet with Next js, I got a `Cannot read property 'extend' of undefined` exception on the server side when using the `FormField` component.

Here is part of my `_document.tsx` file:
```ts
import { Grommet } from "grommet";
import { grommet } from "grommet/themes";

// Inside the render method:

        <body>
          <Grommet theme={grommet}>
            <Main />
          </Grommet>
          <NextScript />
        </body>
```

In a page:

```ts
import React from "react";
import { FormField, TextInput } from "grommet";

export default () => (
  <div>
    <FormField component={TextInput} label="foo">
      <TextInput />
    </FormField>
  </div>
);
```

Will automatically raise the above exception.

Notice that setting styled component doesn't solve anything here.

### Your Environment

- Grommet version: 2.7.11
- Operating System and version (desktop or mobile): Mac OS, Desktop

**Comment by ShimiSun** [comment]
@gaku-sei , hi buddy, can you provide a sandbox that reproduces the problem?

**Comment by oorestisime** [comment]
Out of a hunch @gaku-sei are you using a custom theme? maybe you don't deep merge with base?

**Comment by ShimiSun** [comment]
we released a new version of grommet (2.8.1) couple of days ago with typescript fixes, please let us know if this is still reproducible.

**Comment by ShimiSun** [comment]
Hi buddy, we've released 2.9.0 and I think this one is already fixed. feel free to reopen or file a new issue if it is still reproducing for you.

**Comment by andersondarioo** [comment]
I've created an app with the grommet-starter-new-app and I've got this same error with this code:

`
import React, { Fragment, useState } from 'react';

import { Header, Main, Form, FormField } from 'grommet'

const App = () => {
    const [localidadeOrigem, setLocalidadeOrigem] = useState('')

    return (
        <Fragment>
            <Header>
                Test App
            </Header>

            <Main>
                <Form onSubmit={({ value }) => console.log("Submit: ", value)}>
                    <FormField name="name" label="Name" />
                </Form>
            </Main>
        </Fragment>
    )
}

export default App;

`
